workspaces/jupyterlab-python/Dockerfile

```diff
@@ -17,4 +17,4 @@ RUN /bin/bash /install.sh
 COPY jupyter_server_config.py /etc/jupyter/
 
 USER jovyan
-CMD ["pl-gosu-helper.sh", "start.sh", "jupyter", "lab"]
+ENTRYPOINT ["tini", "-g", "--", "pl-gosu-helper.sh", "start.sh", "jupyter", "lab"]
```

Comment:
Is the intention that you can't override this command anymore (RUN vs ENTRYPOINT)?

Also, is the point of tini to forward signals to the script? It would be nice to have a comment clarifying why tini.

Reply:
It's really CMD vs. ENTRYPOINT. (RUN is for build-time instructions.) ENTRYPOINT is the primary entrypoint and CMD (optional) stacks on top of it. Changing ENTRYPOINT resets CMD to null. What I've done here is replace the old ENTRYPOINT+CMD combo with _only_ an ENTRYPOINT, still keeping the spirit of what Jupyter's upstream was trying to do but injecting our `pl-gosu-helper.sh` into it.

`tini` does several things but that's also something Jupyter chose, not me. (I would probably have chosen it anyway. It's also something that Docker [integrates](https://docs.docker.com/reference/cli/docker/container/run/#init) optionally. It was in use in the previous version before the breaking change too.